src/server/scripts/Outland/zone_hellfire_peninsula.cpp

```diff
@@ -420,10 +420,467 @@ class npc_fel_guard_hound : public CreatureScript
+
+                                events.ScheduleEvent(EVENT_BARADAS_TALK, 10000);
+                                step++;
+                                break;
+                            case 17:
+                                Talk(SAY_BARADA_7);
+                                events.ScheduleEvent(EVENT_BARADAS_TALK, 10000);
+                                step++;
+                                break;
+                            case 18:
+                                if (jules)
+                                    jules->AI()->Talk(SAY_JULES_3);
+
+                                events.ScheduleEvent(EVENT_BARADAS_TALK, 10000);
+                                step++;
+                                break;
+                            case 19:
+                                Talk(SAY_BARADA_7);
+                                events.ScheduleEvent(EVENT_BARADAS_TALK, 10000);
+                                step++;
+                                break;
+                            case 20:
+                                if (jules)
+                                {
+                                    jules->GetAI()->DoAction(ACTION_JULES_MOVE_HOME);
+                                    jules->RemoveAura(SPELL_JULES_VOMITS_AURA);
+                                }
+
+                                events.ScheduleEvent(EVENT_BARADAS_TALK, 10000);
+                                step++;
+                                break;
+                            case 21:
+                                //End
+                                if (playerGUID)
+                                {
+                                    if (Player* player = ObjectAccessor::FindPlayer(playerGUID))
+                                        player->KilledMonsterCredit(NPC_COLONEL_JULES, 0);
+                                }
+
+                                if (jules)
+                                    jules->RemoveAllAuras();
+
+                                me->RemoveAura(SPELL_BARADAS_COMMAND);
+                                me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PACIFIED);
+
+                                Talk(SAY_BARADA_8);
+                                me->GetMotionMaster()->MoveTargetedHome();
+                                EnterEvadeMode();
+                                break;
+                        }
+                        break;
+                }
+            }
+        }
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_baradaAI(creature);
+    }
+};
+
+class npc_colonel_jules : public CreatureScript
+{
+public:
+    npc_colonel_jules() : CreatureScript("npc_colonel_jules") { }
+
+    struct npc_colonel_julesAI : public ScriptedAI
+    {
+        npc_colonel_julesAI(Creature* creature) : ScriptedAI(creature), summons(me) { }
+
+        EventMap events;
+        SummonList summons;
+
+        uint8 circleRounds;
+        uint8 point;
```

Comment:
No, you need to initialize them in the constructor, for example like at https://github.com/TrinityCore/TrinityCore/commit/53d3becf3c5bbefa0d078379decbbc76b2840996